src/ReleaseTests/ProtocolConventionsTest.class.st

```diff
@@ -30,26 +30,83 @@ Some classes are violating this convention:
 								separatedBy: [ s cr ] ])} ]
 ]
 
+{ #category : #asserting }
+ProtocolConventionsTest >> assertProtocolName: aProtocolName notAcceptingProtocolNamesLike: aCollectionOfSelectors notMoreViolationsThan: aCount [
+	| violations |
+	violations := self class environment allClassesAndTraits
+		select: [ :c | (c protocols includesAny: aCollectionOfSelectors) or: [ c class protocols includesAny: aCollectionOfSelectors ] ].
+	self
+		assert: violations size <= aCount
+		description: [ 'In the default Pharo images, the protocol #{1} should be used instead of {2}.
+Some classes are violating this convention:
+{3}'
+				format:
+					{aProtocolName . aCollectionOfSelectors asString . (String
+						streamContents: [ :s | 
+							violations
+								do: [ :c | 
+									s
+										<< '- ';
+										print: c ]
+								separatedBy: [ s cr ] ])} ]
+]
+
+{ #category : #utilities }
+ProtocolConventionsTest >> checkIdiomForRuleClass: aRuleClass [
+
+	self checkIdiomForRuleClass: aRuleClass notMoreViolationsThan: 0
+]
+
+{ #category : #utilities }
+ProtocolConventionsTest >> checkIdiomForRuleClass: aRuleClass notMoreViolationsThan: aCount [
+
+	| idiom |
+	idiom := aRuleClass protocolIdiom.
+	
+	self
+		assertProtocolName: idiom key
+		notAcceptingProtocolNamesLike: idiom value
+		notMoreViolationsThan: aCount
+]
+
+{ #category : #tests }
+ProtocolConventionsTest >> testProperAccessingProtocolIsUsed [
+
+	self checkIdiomForRuleClass: ReProperMethodProtocolNameForAccessingRule notMoreViolationsThan: 1
+]
+
+{ #category : #tests }
+ProtocolConventionsTest >> testProperConvertingProtocolIsUsed [
+
+	self checkIdiomForRuleClass: ReProperMethodProtocolNameForConvertingRule notMoreViolationsThan: 29
+]
+
 { #category : #tests }
 ProtocolConventionsTest >> testProperFileInOutProtocolIsUsed [
 
-	self
-		assertProtocolName: #'file in/out'
-		notAcceptingProtocolNamesLike: #( #'filein/out' #'fileIn/Out' #'fileIn/out' #'filein/Out' )
```

Comment:
Want to add #'file-in/out' ? That's the one I'd use intuitively

Reply:
OK - done in https://github.com/pharo-project/pharo/pull/10171/commits/16a0428f17b8de17421dbf12b82b90b6bd535498

Side note: The "file in/out" was choosen because it is used in image and seems to fit with other "file xxx" patterns used

![image](https://user-images.githubusercontent.com/5980033/143555497-9eaae9ff-90ab-44c9-bf45-b826959ae01e.png)